backpack/extensions/firstorder/variance/rnn.py

```diff
@@ -16,5 +18,21 @@ def __init__(self):
         )
 
     @staticmethod
-    def _get_axis_batch() -> int:
-        return 1
+    def _get_axis_batch(module: RNN) -> int:
+        return 0 if module.batch_first else 1
+
+
+class VarianceLSTM(VarianceBaseModule):
+    """Extension for LSTM, calculating variance."""
+
+    def __init__(self):
+        """Initialization."""
+        super().__init__(
+            params=["bias_ih_l0", "bias_hh_l0", "weight_ih_l0", "weight_hh_l0"],
+            grad_extension=GradLSTM(),
+            sgs_extension=SGSLSTM(),
+        )
+
+    @staticmethod
+    def _get_axis_batch(module: LSTM) -> int:
+        return 0 if module.batch_first else 1
```

Comment:
Use [`get_batch_axis`](https://github.com/tjs111/backpack/blob/458c32038255fdfc9d830570e286b2e9956218c3/backpack/utils/subsampling.py#L28-L65). Same for other occurrences.

Reply:
Done.